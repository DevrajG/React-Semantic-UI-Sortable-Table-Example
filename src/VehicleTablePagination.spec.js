import React from 'react';
import { shallow, mount } from 'enzyme';
import VehicleTablePagination from './VehicleTablePagination';
import { spy } from 'sinon';

describe('VehicleTablePagination', () => {
  it('snapshot regression', () => {
    let onChangePageStub = spy();

    const component = shallow(
      <VehicleTablePagination
        onChangePage = { onChangePageStub }
        totalPages = { 10 }
        currentPage = { 1 }
        showPages = { 3 }
      />);

    expect(component).toMatchSnapshot();
  });


  // it('onSubmitFilter()', () => {
  //   let onChangePageStub = spy();
  //
  //   let wrapper = mount(
  //     <VehicleTablePagination
  //       onChangePage = { onChangePageStub }
  //       totalPages = { 10 }
  //       currentPage = { 1 }
  //       showPages = { 3 }
  //     />
  //   );
  //
  //   expect(vehicleFilter.exists('input')).toEqual(true);
  //   let input = vehicleFilter.find('input');
  //
  //   input.simulate('change', { target: { value: '123' }});
  //   expect(input.instance().value).toEqual('123');
  //   expect(vehicleFilter.state().filterValid).toEqual(true);
  //   expect(onSubmitFilterStub.callCount).toEqual(1);
  //
  //   input.simulate('change', { target: { value: '!@#$' }});
  //   expect(input.instance().value).toEqual('!@#$');
  //   expect(vehicleFilter.state().filterValid).toEqual(false);
  //   expect(onSubmitFilterStub.callCount).toEqual(1);
  // });
});