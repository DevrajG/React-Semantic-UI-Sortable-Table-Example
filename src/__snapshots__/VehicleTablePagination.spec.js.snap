// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleTablePagination snapshot regression 1`] = `
<TableRow
  as="tr"
  cellAs="td"
>
  <TableHeaderCell
    as="th"
    colSpan="8"
  >
    Total pages: 
    10
    <Menu
      borderless={true}
      floated="right"
      pagination={true}
    >
      <MenuItem
        as="a"
        disabled={true}
        icon={true}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="angle double left"
        />
      </MenuItem>
      <MenuItem
        as="a"
        disabled={true}
        icon={true}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="angle left"
        />
      </MenuItem>
      <MenuItem
        active={true}
        as="a"
        key="1"
        onClick={[Function]}
      >
        1
      </MenuItem>
      <MenuItem
        active={false}
        as="a"
        key="2"
        onClick={[Function]}
      >
        2
      </MenuItem>
      <MenuItem
        active={false}
        as="a"
        key="3"
        onClick={[Function]}
      >
        3
      </MenuItem>
      <MenuItem
        active={false}
        as="a"
        key="4"
        onClick={[Function]}
      >
        4
      </MenuItem>
      <MenuItem
        as="a"
        disabled={false}
        icon={true}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="angle right"
        />
      </MenuItem>
      <MenuItem
        as="a"
        disabled={false}
        icon={true}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="angle double right"
        />
      </MenuItem>
    </Menu>
  </TableHeaderCell>
</TableRow>
`;
